### ISSUE TITLE
Query Loop Block: Continued iteration on bundled patterns

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#49280]
>We need a query pattern of a grid of featured images as a starting point. We need much more optionality on query patterns in general.

Pulled from https://github.com/WordPress/gutenberg/issues/48443 and a follow up to https://github.com/WordPress/gutenberg/issues/44140

cc @richtabor  in case you have time to work on this.

**Comment by richtabor** [comment]
I'll see what I can do, though last I experimented the blockGap limitation (https://github.com/WordPress/gutenberg/issues/44899) was making it difficult to get spacing right in a grid layout.

**Comment by tellthemachines** [comment]
I'm thinking #49050 might be helpful to move this forward, but it needs a bit of input on how we want the columns behaviour in the grid template to work. The V1 of grid layout only allows setting a minimum column width, not a number of columns, but has the advantage of better responsivity out of the box vs the current solution. Using the grid layout will also allow us to easily add block spacing, which would fix #44899.

**Comment by richtabor** [comment]
> Using the grid layout will also allow us to easily add block spacing, which would fix https://github.com/WordPress/gutenberg/issues/44899.

Would you have to use grid layout to modify block spacing then?

**Comment by richtabor** [comment]
> Using the grid layout will also allow us to easily add block spacing, which would fix https://github.com/WordPress/gutenberg/issues/44899.

Post Template should probably have block gap too, as its children will inherit that parent block's spacing value — not the query loop's. 

> Using the grid layout will also allow us to easily add block spacing, which would fix https://github.com/WordPress/gutenberg/issues/44899.

As long as using grid layout isn't required for block spacing support.

**Comment by mtias** [comment]
What's the status here? Is there anything actionable remaining?

**Comment by annezazu** [comment]
Yes, we need to incorporate more patterns still! Room for improvement remains.

**Comment by annezazu** [comment]
Noting that another pattern that's needed is one that shows everything: post titles, full post content, tags, categories, author. Adding this pattern would be the fastest way to set up a blog as your homepage.
